src/routes/app/upload/+layout.ts

```diff
@@ -0,0 +1,14 @@
+/**
+ * Data loading for upload
+ */
+
+import * as projectsApi from "$lib/api/projects.js";
+
+export async function load({ fetch }) {
+  const [pinned, projects] = await Promise.all([
+    projectsApi.list({ pinned: true }, fetch),
+    projectsApi.list({ per_page: 100 }, fetch),
```

Comment:
From what I can tell, this `list` method will return all public projects matching the provided params, including projects the user does not have edit access for. I think this should instead be a list of projects where the logged-in user is the project `user` or one of its `collaborators`.